app/controllers/comments_controller.rb

```diff
@@ -1,19 +1,30 @@
 class CommentsController < ApplicationController
-  load_and_authorize_resource
   before_action :find_user
   before_action :find_post
+  before_action :set_user_for_comment
+
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.all
+
+    render json: @comments
+  end
 
   def new
     @comment = Comment.new
   end
 
   def create
     @comment = Comment.new(comment_params)
-    @comment.author = @user
+    @comment.author = @c_user
     @comment.post = @post
+
     if @comment.save
       flash[:notice] = 'Comment created successfully.'
-      redirect_to user_post_path(@user, @post)
+      respond_to do |format|
+        format.html { redirect_to user_post_path(@user, @post) }
+        format.json { render json: @comment, status: :created }
+      end
     else
       render 'new'
     end
```

Comment:
- [Optional] Dear Team good job on the project. Also, I would suggest having a  `format.json` with an error code whenever creating or finding a comment/post fails. This will ensure that even if the app fails, the response will always be sent to the end user.